src/visual.ts

```diff
@@ -989,7 +1018,7 @@ module powerbi.extensibility.visual {
             if (this.data.settings.labels.show) {
                 const labelsXScale: LinearScale<number, number> = d3.scale.linear()
                     .domain([0, series[0].dataPoints.length - 1])
-                    .range([0, width - margin.left - margin.right]);
+                    .range([0, width - margin.left - (this.margin.right + this.data.xAxisValueMaxSize / 2)]);
```

Comment:
Please simplify all lines like this one: width - margin.left - (this.margin.right + this.data.xAxisValueMaxSize / 2)

Reply:
Also I don't see any unit-tests here